cms/server/admin/handlers/contestannouncement.py

```diff
@@ -60,6 +60,40 @@ def post(self, contest_id: str):
                 make_datetime(), "Subject is mandatory.", "")
         self.redirect(self.url("contest", contest_id, "announcements"))
 
+class EditAnnouncementHandler(BaseHandler):
+    """Called to edit an announcement"""
+
+    @require_permission(BaseHandler.PERMISSION_MESSAGING)
+    def post(self, contest_id: str, ann_id: str):
+        ann = self.safe_get_item(Announcement, ann_id)
+        self.contest = self.safe_get_item(Contest, contest_id)
+
+        # Protect against URLs providing incompatible parameters.
+        if self.contest is not ann.contest:
+            raise tornado.web.HTTPError(404)
+
+        self.sql_session.delete(ann)
+        self.try_commit()
```

Comment:
You shouldn't delete the announcement if there are errors (i.e. the subject is empty)